### DISCUSSION TITLE
Lesson 4: Ganache mismatch between chain and network ID

### DISCUSSION THREAD

**Discussion by pythoulon** [smartcontractkit/full-blockchain-solidity-course-py#111] [Q&A] [answered]
Lesson 4 says (unless I misunderstood) that we should use the "Network ID" reported by Ganache as chain ID for the transaction. In my case, Ganache reports a Network ID of 5777 (see screen capture below), but if I use that, web3.py complains when I run deploy.py with the following message:

> web3.exceptions.ValidationError: The transaction declared chain ID 5777, but the connected node is on 1337

When I set my chain ID to 1337, everything goes fine and deploy.py executes properly. Any idea why I have this behavior ? Searched the documentation and the web in general to no avail...

Thanks.

![image](https://user-images.githubusercontent.com/11643101/135035028-7ac99350-7037-4a7c-86bd-f23106722630.png)

**Comment by 350z195** [comment, marked as answer]
I ran into the same problem, what I did was to go into the settings of ganache->server->network ID in which I changed my network ID to 1337. This will change your address and private key so make sure to update those as well in your code.

**Reply by pythoulon** [reply]
Glad to see it's not just me 😁. Seems like a problem with Ganache. Anyway, it was not really blocking. Thanks for your reply.

**Reply by PatrickAlphaC** [reply]
Awesome work hear all! Great Q&A

**Reply by JamiDEV** [reply]
I had the same issue, thanks for the update.

**Reply by marginalfriend** [reply]
I also ran into the same issue, but I fixed it by only changing the chain_id to 1337 as the machine wished and not change the server settings, will that cause any other problem?
